### DISCUSSION TITLE
Ideas for a debugging extension for OpenMage

### DISCUSSION THREAD

**Discussion by elidrissidev** [OpenMage/magento-lts#3219] [Ideas]
I had an idea for a while to create an extension for OpenMage to help with debugging. So I was able to get some time today to create a small prototype and I'd like to get your help for ideas on what kind of information would be useful to add to it. The extension uses [php-debugbar](https://github.com/maximebf/php-debugbar/) which adds an expandable bar fixed to the bottom of the page with tabs containing different information.

Some ideas on my head:

- Logs: shows all log messages that were logged in the last request
- Db: shows all database queries that were executed during the last request. It's already working, but missing metrics like memory usage
- Config: shows config tree for the current store
- Route: shows information about the current route and controller
- PHP info: currently showing only version, maybe it can also show enabled extensions and php.ini configuration?

I imaging some of these features will be a pain to implement without overwriting lots of files but let's see 😅

<img width="1439" alt="Screen Shot 2023-04-25 at 1 20 22 PM" src="https://user-images.githubusercontent.com/67818913/234274245-b2d595f0-9789-43ec-a649-2dfe71318416.png">

**Comment by fballiano** [comment]
how would it compare to something like https://github.com/magespecialist/m1-MSP_DevTools? maybe there's something that can be borrowed from that extension?

**Comment by fballiano** [comment]
what I'd really need would be something like blackfire without having to pay 2 kidneys for that :-\

**Reply by elidrissidev** [reply]
FYI [Prometheus](https://prometheus.io/) is an alternative that is open-source. I wouldn't go that far into making the extension an APM tool but something like the database queries view should help a bit.

**Reply by fballiano** [reply]
counting the repeating database queries, with timing, already makes the world of difference

**Reply by Flyingmana** [reply]
https://pyroscope.io/ is an equivalent

Prometheus is a more general metrics Collector and UI and maybe not matching what blackfire provides.

**Reply by fballiano** [reply]
wow pyroscope seems to be exactly that! and it has a few tier that's actually usable!

**Reply by sreichel** [reply]
> what I'd really need would be something like blackfire without having to pay 2 kidneys for that :-\

Erm ... blackfire.io is listed on partners page (https://www.openmage.org/partners/). I got a free account for OM some years ago (but never really used it). Just get in contact?
